manifests/install.pp

```diff
@@ -18,17 +18,43 @@
     if $::operatingsystem != 'darwin' {
       ensure_packages(['unzip'])
     }
-    staging::file { 'consul.zip':
-      source => $consul::download_url
-    } ->
-    staging::extract { 'consul.zip':
-      target  => $consul::bin_dir,
-      creates => "${consul::bin_dir}/consul",
-    } ->
-    file { "${consul::bin_dir}/consul":
-      owner => 'root',
-      group => 0, # 0 instead of root because OS X uses "wheel".
-      mode  => '0555',
+
+    # Installing consul for the very first time
+    if ($::consul_version == undef) {
+
+      staging::file { "consul_${consul::version}.zip":
+        source => $consul::download_url,
+      } ->
+      staging::extract { "consul_${consul::version}.zip":
+        target => $consul::bin_dir,
+      } ->
+      file { "${consul::bin_dir}/consul":
+        owner => 'root',
+        group => 0, # 0 instead of root because OS X uses "wheel".
+        mode  => '0555',
+      }
+
+    # Upgrading existing installation of consul
+    } elsif ($::consul_version != $consul::version) {
```

Comment:
This seems like an iffy way to do this. Puppet works better in a kind of "tell don't ask" kind of way.

Can you think of a better way to do this?

Heck what if we just did this?

```puppet
file { '/etc/consul_version': content => $consul::version } ~>
exec { "rm -rf bin/consul": refreshonly => true  }
```